### PR TITLE
Update secret name for backup-restic addon to backup-restic-s3-credentials

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
Signed-off-by: Waleed Malik <ahmedwaleedmalik@gmail.com>

**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
Currently, there is a conflict between the `s3-credentials` secret created by KKP and KubeOne. Both use the same secret name and deploy it in the same namespace i.e. kube-system. But the keys within the secret are different for both of them.

In a bid to solve this issue, we are renaming the secret in KubeOne.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
[ACTION REQUIRED] Update secret name for backup-restic addon to kubeone-backups-credentials. Manual migration steps are needed for users running KKP on top of a KubeOne installation and using both backup-restic addon from KubeOne and s3-exporter from KKP. Ensure that the `s3-credentials` Secret with keys `ACCESS_KEY_ID` and `SECRET_ACCESS_KEY` exists in `kube-system` namespace and doesn't have the label `kubeone.io/addon:`. Remove the label if it exists. Else s3-exporter won't be functional.
```

### CODE REVIEW

File: addons/backups-restic/backups-restic.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 kind: Secret
 metadata:
-  name: s3-credentials
+  name: backup-restic-s3-credentials
```
Comment (ahmedwaleedmalik) [inline addons/backups-restic/backups-restic.yaml:4]: @kron4eg @xmudrii This or more specific `kubeone-backup-restic-credentials`?

Comment (xmudrii) [reply addons/backups-restic/backups-restic.yaml:4]: I was planning to go with `kubeone-backups-credentials`.

Comment (ahmedwaleedmalik) [reply addons/backups-restic/backups-restic.yaml:4]: That sounds fine as well, my intention behind not adding the `kubeone-` prefix but we already use that prefix for other credentials. So let's do the same here.

Comment (ahmedwaleedmalik) [reply addons/backups-restic/backups-restic.yaml]: Updated PTAL

**Comment by xmudrii** [comment]
@ahmedwaleedmalik Can you please add `[ACTION REQUIRED]` release note and mention that users using both this addon and s3-exporter should take additional steps?

**Comment by ahmedwaleedmalik** [comment]
@xmudrii Do they really have to? As in when they'll update kubeone, the new dedicated secret will be used for this addon. And the de-coupling would be removed.

Same will happen if they update KKP first.

**Comment by ahmedwaleedmalik** [comment]
Are you talking about the case where a K1 update would prune/remove the existing `s3-credentials`?

**Comment by ahmedwaleedmalik** [comment]
Are you talking about the case where a K1 update would prune/remove the existing `s3-credentials`?

**Comment by xmudrii** [comment]
@ahmedwaleedmalik Yes, I was talking about pruning the s3-credentials Secret.

File: pkg/kubeconfig/kubeconfig.go [hunk 2]
```diff
@@ -76,18 +76,10 @@ func BuildKubernetesClientset(s *state.State) error {
 
 	s.RESTConfig.Dial = tunn.TunnelTo
 
-	return errors.WithStack(HackIssue321InitDynamicClient(s))
-}
-
-// HackIssue321InitDynamicClient initialize controller-runtime/client
-// name comes from: https://github.com/kubernetes-sigs/controller-runtime/issues/321
-func HackIssue321InitDynamicClient(s *state.State) error {
```
Comment (kron4eg) [inline pkg/kubeconfig/kubeconfig.go:84]: This function is a reminder and explanation of "why it's done this way", please don't remove it

Comment (ahmedwaleedmalik) [reply pkg/kubeconfig/kubeconfig.go:84]: It's not required anymore and actually results in a nil pointer exception if k8s was not properly installed.

**Review by kron4eg — APPROVED** [review]

**Pull request merged**